okta/table_okta_password_policy.go

```diff
@@ -83,13 +146,27 @@ func listPolicies(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateDat
 
 	// paging
 	for resp.HasNextPage() {
-		var nextPolicySet []*okta.Policy
+		var nextPolicySet []*PolicyStructure
 		resp, err = resp.Next(ctx, &nextPolicySet)
 		if err != nil {
-			logger.Error("listPolicies", "list_policies_with_settings_paging_error", err)
+			logger.Error("listOktaPoliciesIdentityEngine", "list_policies_paging_error", err)
 			return nil, err
 		}
 		for _, policy := range nextPolicySet {
+			// Additional API call to get rules for Identity Engine
+			rules, _, err := client.Policy.ListPolicyRules(ctx, policy.Id)
+			if err != nil {
+				logger.Error("listOktaPoliciesIdentityEngine", "list_policy_rules_error", err)
+				return nil, err
+			}
+			// Ensure policy.Embedded is a map[string]interface{}
+			if policy.Embedded == nil {
+				policy.Embedded = make(map[string]interface{})
+			}
+			// Embed the rules
+			policyEmbedded := policy.Embedded.(map[string]interface{})
+			policyEmbedded["rules"] = rules
```

Comment:
Could you please confirm if this block of code is returning the rules as expected? We are only checking if the `policy.Embedded` value is null or not; we are not assigning any value to it.